client/client.go

```diff
@@ -71,10 +69,10 @@ func (c *Client) Close() error {
 	return c.conn.Close()
 }
 
-type withBlockOpt struct{}
+type withFailFast struct{}
 
-func WithBlock() ClientOpt {
-	return &withBlockOpt{}
+func WithFailFast() ClientOpt {
+	return &withFailFast{}
```

Comment:
nit, but i don't see why this has to be a pointer

Reply:
This could use some normalization but don't want to do this now and leave it different from others.